docs/source/examples/checkpointer.rst

```diff
@@ -0,0 +1,319 @@
+.. _understand_checkpointer:
+
+==============================
+Understanding the Checkpointer
```

Comment:
Do we wanna add one of those "What you will learn"/"Prerequisites" headers (similar to e.g. our "Finetune your First LLM" tutorial)? Might be helpful to (a) standardize our format a bit, and (b) make sure readers have the appropriate context going in

Reply:
Also maybe I am nitpicking too much, but we could also just call this tutorial "Checkpoints in TorchTune". To me "Understanding X" indicates that X is inherently difficult to understand

Reply:
Both good points. I had the "What you will learn" and then removed it. Will bring it back. Good point on the name